### PR TITLE
Feature/howard write nu id features v09 42 00

### PR DESCRIPTION
**Pull request by brucehoward-physics**
**NOTE!!!** Don't merge until some updates from LArPandoraContent are in. Relies on a PR there from Henry that is in progress. But I wanted to start putting in PRs for review.

Including same group of reviewers as in the `sbnanaobj` PR, but please feel free to change/add/remove as necessary.
Depends on https://github.com/SBNSoftware/sbnanaobj/pull/43

### CODE REVIEW

File: sbncode/CAFMaker/FillReco.cxx [hunk 1]
```diff
@@ -273,6 +283,38 @@ namespace caf
...
+        srslice.nuid.NuNHitsTotal = properties.at("NuNHitsTotal");
+      }
+      if (properties.count("NuVertexY")) {
+	srslice.nuid.NuVertexY = properties.at("NuVertexY");
+      }
+      if (properties.count("NuWeightedDirZ")) {
+	srslice.nuid.NuWeightedDirZ = properties.at("NuWeightedDirZ");
+      }
+      if (properties.count("NuNSpacePointsInSphere")) {
+	srslice.nuid.NuNSpacePointsInSphere = properties.at("NuNSpacePointsInSphere");
```
Comment (cjbackhouse) [inline sbncode/CAFMaker/FillReco.cxx:300]: Something up with the indentation (literal tabs?) here

File: sbncode/CAFMaker/FillReco.cxx [hunk 2]
```diff
@@ -273,6 +283,38 @@ namespace caf
       else {
         srslice.nu_score = -1;
       }
+      // NeutrinoID (SliceID) features
+      if (properties.count("NuNFinalStatePfos")) {
+        srslice.nuid.NuNFinalStatePfos = properties.at("NuNFinalStatePfos");
+      }
```
Comment (cjbackhouse) [inline sbncode/CAFMaker/FillReco.cxx:289]: I wonder if we can make some `CopyPropertyIfSet()` helper function to reduce the repetition here.
```suggestion
      CopyPropertyIfSet(properties, "NuNFinalStatePfos", srslice.nuid.NuNFinalStatePfos);
```

File: sbncode/CAFMaker/FillReco.cxx (hunk 1 above)
Comment (brucehoward-physics) [reply sbncode/CAFMaker/FillReco.cxx:300]: Interesting. Pretty sure it looked okay in my editor but wouldn't be the first time indentations were treated weirdly in an editor 🤦‍♂️. I'll touch that up.

**Review by henrylay97 — APPROVED** [review]
Looks good to me (with one suggestion)! 

I have tested this set of PRs with an `sbndcode` setup (as I know Bruce did ICARUS) and successfully re ran reco2 & caf on a file. The plots of the variables look sensible to me having worked with them for the last few weeks.

For those less familiar with pandora it is worth pointing out that over 50% of the time these variables will be unfilled because they are not calculated for unambiguous cosmic slices. If you want to view a useful plot use the condition `!rec.slc.is_clear_cosmic`.

**Comment by pgreen135** [comment]
@brucehoward-physics this (and the corresponding sbnanaobj PR) appear to be based off version v09_42_00; which means they cannot be merged directly into the sbn2021c/2022a release without also merging undesired changes from develop. Can you remake this based off the correct tag? (v09_37_01_03) -- (unless you are planning to manually do this @miquelnebot?)

File: sbncode/CAFMaker/FillReco.cxx [hunk 3]
```diff
@@ -255,6 +255,16 @@ namespace caf
     // default values
     srslice.nu_score = -1;
     srslice.is_clear_cosmic = true;
+    srslice.nuid.NuNFinalStatePfos = -9999.f;
+    srslice.nuid.NuNHitsTotal = 0;
```
[the remark this answers is not in the document]
Comment (brucehoward-physics) [reply sbncode/CAFMaker/FillReco.cxx:259]: Yeah this falls under the category I mentioned in the other PR that default values for unsigned ints are a little awkward. I could make it set to the max unsigned int or something but not sure that feels great either?

[the remark this answers is not in the document]
Comment (cjbackhouse) [reply sbncode/CAFMaker/FillReco.cxx:259]: Do they really need to be unsigned? I understand that they are logically counts, but letting them be signed gives us a convenient -1 == invalid value.

[the remark this answers is not in the document]
Comment (brucehoward-physics) [reply sbncode/CAFMaker/FillReco.cxx:259]: That's fair - I guess it's unlikely that we need the unsignedness. We're converting it from a double in the map back to an integer type in any case so yeah I think I can just make it a signed int. I'm going to be resubmitting the PRs (probably today) in v09_37 series, so I will make that change for that PR.

**Comment by brucehoward-physics** [comment]
Closing to remake PR in v09_37 series.

**Pull request closed without merging**